### PR TITLE
VUE-512 live loan images, loan-to-loan

### PR DESCRIPTION
**Pull request by ryan-ludwig**
Adds ability to hit loan-to-loan recommendation urls like 
`/live-loan/l/*loanid*/img/1` &
`/live-loan/l/*loanid*/url/1`

Retains loginid recommendation urls like
 `/live-loan/u/*loginid*/img/1` &
`/live-loan/u/*loginid*/url/1`
 
Refactor to reduce duplicate code.

### CODE REVIEW

**Comment by mcstover** [comment]
Could you please create a util method (could be in this file for now) to format all console logs or errors with this structure? 

Just switch between info and error where appropriate.

`console.info/error/warning(JSON.stringify({
    meta: {},
    level: 'info', // info or error or warning
    message: ###message###
})));`

File: server/live-loan-router.js [hunk 1]
```diff
@@ -3,170 +3,201 @@ const express = require('express');
...
+			await memJsUtils.setToCache(
+				`recommendations-by-${type}-id-${id}`,
+				JSON.stringify(loanData),
+				expires,
+				cache
+			);
+			return loanData;
+		} catch (err) {
+			// throw if we can't save to memcache since not using caching would be problematic during an email campaign
+			throw new Error(`Error setting loan data to cache: ${err}`);
```
Comment (ryan-ludwig) [inline server/live-loan-router.js:131]: Should we error out if we can't save to memcache here?

**Comment by ryan-ludwig** [comment]
Would love to get a few eyes on this since quite a bit was refactored.

File: server/live-loan-router.js (hunk 1 above)
Comment (mcstover) [reply server/live-loan-router.js:131]: We should initially format and log the error for now...(on line 115 too)

A future improvement would be to return the results of a generic loan set for both the image and url states.

File: server/live-loan-router.js [hunk 2]
```diff
@@ -3,170 +3,203 @@ const express = require('express');
...
+				expires,
+				cache
+			);
+			return loanData;
+		} catch (err) {
+			log(`Error setting loan data to cache, ${err}`, 'error');
+		}
+	} else {
+		// future improvement, return a default set of 4 loans here
+		throw new Error('No loans returned');
```
Comment (ryan-ludwig) [inline server/live-loan-router.js:134]: @mcstover I'm throwing here to reject out of the promise, which gets caught and logged in `redirectToUrl` and `serveImg`. An alternative is to resolve as a blank array, and then check that we have loans before processing on lines 143 and 159.

**Review by mcstover — APPROVED** [review]

**Pull request merged**